### ISSUE TITLE
Firmware help

### ISSUE THREAD

**Issue by shade1059** [revarbat/snappy-reprap#33]
Are there instructions on how to flash the firmware? I am new to marlin. Specifically the LCD screen on the printer says "printer ready" but nothing else, and I do not know where to begin with troubleshooting.

**Comment by ManuGithubSteam** [comment]
See here for example: https://www.my-home-fab.de/en/documentations/reprap-firmware/installing-marlin-firmware-on-ramps-1.4
Just leave out step 3 as the firmware is already modifierd for the snappy. So just dont do the modifications.

**Comment by shade1059** [comment]
Thank you for the help. The Arduino software tells me that it compiled and flashed correctly, however the LCD screen comes up blank, and my computer fails to connect every time. Through some fiddling with the marlin settings, I can get the LCD screen to say "printer ready 16/0" but it is unresponsive beyond that.

**Comment by ManuGithubSteam** [comment]
Hmm things that pop into mind:

- wirering issue
- you fried some part of the mega or ramps ( these boards are so cheap they
have no protection for " upps wire should go here not here)

- what fiddling do you do to get the LCD to work?
- did you try print with octoprint? Maybe LCD just borked

**Comment by shade1059** [comment]
-I looked through my wiring and couldn't find an issue.
-How would you know if you fried part of the mega? I was told that the LCD would go haywire if you did, and I have not yet seen that.

-I have not tried octoprint yet, however I did plug it into my computer and used simplify 3d to control it directly. It will connect to the printer and display the nozzle temperature accurately. If I try to move any motors or activate any heating elements, the machine will not respond. In the case of the heating elements, the software will eventually report a fault due to heating element temperature not changing.
I figure if I messed up the wiring somewhere, this should give me a hint. Also figured that since the temperature readout was correct to room temperature, and if I breathed on the nozzle, the temperature went up, this indicates I probably did not fry any electronics

-to get the LCD to work, uncomment line 1322 "#define ULTRA_LCD   // Character based", comment out line 1494 "#define REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER" I also changed line 122 to   "#define MOTHERBOARD 43" But I think I was supposed to do that originally, the instructions were just not clear on that.


Note, all line changes are in the configuration.h tab

**Comment by ManuGithubSteam** [comment]
Hi Shade 

i have good news.
I was able to compile the stock marlin firmware and get it to work on the ramps 1.4.
I uncommented the lines you mentioned. But i did NOT touch the motherboard line.
I strongly believe, you have to set it to Ramps SOMETHING (can't access config right now...)

On mine it was: BOARD_RAMPS_13_EEB as the ramps support heated bed and two extruders.

I think this is needed to make the right compiler flags

Then i flashed it and the lcd did came on and worked with the buttons etc.
However i did not connect the motors and switches etc at that time.

Hardware i use: Mega2560 + Ramps 1.4 

Sadly i was not able to compile the snappy marlin Firmware. It gave me an error with the y Endstop.
Did you successful compile the snappy version of the firmware ?

If so, would you upload your sources here ?
Do you know if the snappy firmware does work with the Endstops switched from top to bottom ? (3.1 design)

As said the stock marlin does compile and flash onto the ramps. My Plan for now is to disect the config.h from the marlin and grad usefull stuff while hopefully let it compile and see where i go fro there...

Thanks for the help!

**Comment by ManuGithubSteam** [comment]
Ok here it is. My first Marlin Firmware for snappy 3.1.

What did i do:
- basicly comb trough the orginal snappy config and copy all the sensible lines into the new Marlin config.h
- I did enabled a few nice things that are not in original snappy but i tried to keep it simple and not mess up all.
- I did not flip the z axis as it was switched back to the bottom of the z tower in 3.1 design.

It compiles.
So please test it. This is really alpha. I did not even test it on the device. just checked if it compiles!

Basically i think it will work but most likely needs some tinkering.
Please test.

And remember to set your correct motherboard in the config.h
[Marlin_Snappy_Next_v1.tar.gz](https://github.com/revarbat/snappy-reprap/files/2809341/Marlin_Snappy_Next_v1.tar.gz)

**Comment by ManuGithubSteam** [comment]
Here is my semi final version.
- Axis go in the correct direction (wired like 3.0)
- Endstops do work.
- Soft Endstops work

Over all this should print quite well. I have some issues with the z rods as they block if they are not 100% flat.
I did not test real printing yet as i have the above issue.


[Marlin_Snappy_Next.zip](https://github.com/revarbat/snappy-reprap/files/2833322/Marlin_Snappy_Next.zip)

**Comment by ManuGithubSteam** [comment]
Ok here is a working version with filament unload etc. I printed a benchy with it

[Marlin_Snappy_Next_v7.zip](https://github.com/revarbat/snappy-reprap/files/2873146/Marlin_Snappy_Next_v7.zip)

**Comment by ManuGithubSteam** [comment]
Here is a new verision with curve cuttent to the steppers. Reduces vibrations quite a bit i think:
[Marlin_Snappy_Next_v8.zip](https://github.com/revarbat/snappy-reprap/files/2912245/Marlin_Snappy_Next_v8.zip)

**Comment by shade1059** [comment]
Sorry for the delayed response. I was able to get it to work eventually. There were a few issues: The LCD shipped didnt exactly match what I ordered, and I didnt notice. Thus it didnt quite align with the model enabled in the firmware.
Also my power supply was having issues, I ended up getting an new one and all of the motors started working properly. Now I know I have the wrong number of steps per mm on my Z, but just gotta take some simple measurements to fix that (or see what you used in the marlin that you sent me).  I am good with marking this this issue taken care of.
